Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesFragment.java

```diff
@@ -0,0 +1,191 @@
+                    int progress = intent.getIntExtra(GeoTagImagesService.EXTRA_PROGRESS, 0);
+                    updateState(STATE_GEOTAGGING);
+                    progressBar.setMax(total);
+                    progressBar.setProgress(progress);
+                    break;
+            }
+        }
+    };
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+
+        if (!(context instanceof GeoTagActivity)) {
+            throw new IllegalStateException("Activity is not instance of " + GeoTagActivity.class.getSimpleName());
+        }
+
+        activity = (GeoTagActivity) context;
+        lbm = LocalBroadcastManager.getInstance(context);
+
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_geotag_images, container, false);
+    }
+
+    int currState = STATE_INIT;
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        instructionText = (TextView) view.findViewById(R.id.instruction_text);
+        geotagButton = (Button) view.findViewById(R.id.geotag_button);
+        progressBar = (ProgressBar) view.findViewById(R.id.geotag_progress);
+        sdCard = (ImageView) view.findViewById(R.id.sd_card);
+
+        instructionText.setText(R.string.insert_sdcard);
+        geotagButton.setText(R.string.label_begin);
+
+        sdCardAnim = AnimationUtils.loadAnimation(getContext(), R.anim.sd_card_anim);
+
+        updateState(STATE_INIT);
+
+        geotagButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                switch (currState) {
+                    case STATE_INIT:
+                        updateState(STATE_GEOTAGGING);
+                        startGeoTagging();
+                        break;
+                    case STATE_GEOTAGGING:
+                        updateState(STATE_INIT);
+                        cancelGeoTagging();
+                        break;
+                }
+            }
+        });
+
+        if (activity != null) {
+            activity.updateTitle(R.string.geo_tag_label);
+        }
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        lbm.registerReceiver(receiver, filter);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        lbm.unregisterReceiver(receiver);
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+        activity = null;
+    }
+
+    private void startGeoTagging() {
+        Context context = getContext();
+        Intent intent = new Intent(context, GeoTagImagesService.class);
+        intent.setAction(GeoTagImagesService.ACTION_START_GEOTAGGING);
+        getContext().startService(intent);
```

Comment:
@chaviw should ensure `getContext()` is not null, and reuse the `context` variable defined at the beginning of the method.